src/BloomExe/TempFiles.cs

```diff
@@ -119,6 +100,15 @@ public static XmlWriterSettings GetXmlWriterSettingsForHtml5()
 			return settings;
 		}
 
+		// xml output will produce things like <title /> or <div /> for empty elements, which are not valid HTML 5 and produce
+		// weird results; for example, the browser interprets <title /> as the beginning of an element that is not terminated
+		// until the end of the whole document. Thus, everything becomes part of the title. This then causes errors in our
+		// thumbnail generation because gecko thinks the document has an empty  body (the real one is lost inside the title).
+		// Also, embedded controls (like ReaderTools.htm) now pass through this xml-to-html conversion, and this file contains
+		// several more kinds of empty element, some of which have attributes.
+		// There are probably more elements than these which may not be empty. However we can't just use [^ ]* in place of title|div
+		// because there are some elements that never have content like <br /> which should NOT be converted.
+		// It seems safest to just list the ones that can occur empty in Bloom...if we can't find a more reliable way to convert to HTML5.
```

Comment:
(a) This is not a new comment, just moved (in a refactoring that would have been better in its own commit, sorry).
(b) We don't need to list the ones like <br /> which should NOT have closing elements...just the ones which MUST have them, and which Tidy unfortunately contracts. We could possibly add more of those if it seems worth it.
(c) This is one of the bugs we should investigate to see whether it has been fixed in Tidy. But not as part of this PR.